src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -258,9 +264,33 @@ public WorkflowRun(WorkflowJob job, File dir) throws IOException {
             if (definition == null) {
                 throw new AbortException("No flow definition, cannot run");
             }
+
+            boolean loggedHintOverride = false;
+            if (getParent().isResumeBlocked()) {
+                definition.setDurabilityHint(FlowDurabilityHint.PERFORMANCE_OPTIMIZED);
```

Comment:
This is not right. `FlowDefinition`s are `Describable`s. Thus they must never have state that is modified except in direct response to a user saving a configuration screen or the like. Anyway a single definition might in principle be shared among any number of jobs, not to mention builds.